tests/common/widgets/current-refinements/index.ts

```diff
@@ -0,0 +1,30 @@
+import { fakeAct } from '../../common';
+
+import { createOptionsTests } from './options';
+
+import type { TestOptions, TestSetup } from '../../common';
+import type { CurrentRefinementsWidget } from 'instantsearch.js/es/widgets/current-refinements/current-refinements';
+
+type WidgetParams = Parameters<CurrentRefinementsWidget>[0];
+export type CurrentRefinementsWidgetSetup = TestSetup<{
+  widgetParams: Omit<WidgetParams, 'container'>;
+  testParams?: {
+    /**
+     * Sets up an InstantSearch instance wrapped in a form element with the provided `onSubmit` event handler.
+     */
+    formWrapperSubmitHandler?: jest.Mock<any, any>;
+  };
```

Comment:
For a test I needed to have InstantSearch inside of a `<form>` with mocked submit event handler. This new `testParams` can be defined during `setup()` to send additional information to the flavor-specific implementation of the test suite.

Reply:
I just tried it out, and you can attach a "submit" event handler to body, but only when the button is in a form:

```js
// works:
document.body.append(document.createElement('form'))
document.querySelector('form').append(document.createElement('button'), { textContent: 'click me' }))
document.body.addEventListener('submit', console.log)

// does not work
document.body.append(document.createElement('button'), { textContent: 'click me' }))
document.body.addEventListener('submit', console.log)
```

This means we can change the setup to always put the CR in a form, and in the test attach a listener to body's submit event

Reply:
The form is now created by the options test and the InstantSearch implementations use it as their root container: https://github.com/algolia/instantsearch/pull/5838/commits/c519cbb63c2729b12d1a95f9f18e980ea56b308d.